File: algorithms/Strings/Easy/LongestCommonPrefix/VSDLongestCommonPrefix.cpp

#include<iostream>
#include<string>
#include<vector>

using namespace std;


string getMinString(vector<string> &strs){
    string minstring=strs[0];
    for(int i=0;i<strs.size();i++){
        if(strs[i].size()<minstring.size())
            minstring=strs[i];
    
    }
    return minstring;

}

string longestCommonPrefix(vector<string> &strs){
    string answer="";
    int n=strs.size();
    if(n==1)
        return strs[0];
    string minstr=getMinString(strs);
    int m=minstr.size();
    for(int i=0;i<m;i++){
        char ch=strs[0][i];
        for(int j=0;j<n;j++){
            if(ch!=strs[j][i])
                return answer;
        }
        answer+=ch;

    }
    return answer;
}


int main(){
    vector<string> strs{"flower","flow","flight"};
    string ans=longestCommonPrefix(strs);
    cout<<ans;

    return 0;
}

File: algorithms/Strings/Easy/LongestCommonPrefix/VSDLongestCommonPrefix.java


public class Main{

    static String getMinString(String[] strs){
        String minString=strs[0];
        for(String s:strs){
            if(s.length()<minString.length())
                minString=s;
            
        }
        return minString;
    }

    static String longestCommonPrefix(String[] str){
        if(str.length==1)
            return str[0];
        String ans="";
        String minString = getMinString(str);

        for(int i=0;i<minString.length();i++){
            char ch = str[0].charAt(i);

            for(int j=0;j<str.length;j++){
                if (ch != str[j].charAt(i))
                    return ans;
            }
            ans+=ch;
        }
        return ans;

    }
    public static void main(String[] args) {
        String[] strs={"flower","flow","flight"};
        String ans=longestCommonPrefix(strs);
        System.out.println(ans);
    }
}